crates/node/src/lib.rs

```diff
@@ -302,12 +305,23 @@ impl Node {
                 root_hash,
                 artifact,
             } => {
+                let Some(sender_key) = self.ctx_manager.get_sender_key(&context_id, &author_id)?
+                else {
+                    return self.initiate_sync(context_id, source).await;
+                };
+
+                let shared_key = SharedKey::from_sk(&sender_key);
+
+                let artifact = &shared_key
+                    .decrypt(artifact.into_owned(), [0; aead::NONCE_LEN])
+                    .ok_or_eyre("failed to decrypt message")?;
+
                 self.handle_state_delta(
                     source,
                     context_id,
                     author_id,
                     root_hash,
-                    artifact.into_owned(),
+                    artifact.to_vec(),
```

Comment:
you shouldn't need this

```suggestion
                    artifact,
```